### DISCUSSION TITLE
How to provision certificate chains to device

### DISCUSSION THREAD

**Discussion by Mumunn** [DMTF/libspdm#2871] [Q&A] [answered]
Hi, if SPDM Spec: DSP0274_1.1.0 is used, could you please tell me how to provision a device with certificate chains?

**Comment by steven-bellock** [comment, marked as answer]
That information is specific to a device and device vendor. For the slot 0 certificate chain it is usually provisioned during manufacturing, and SPDM itself is not involved. Ie, the certificate chain is sent to the non-volatile storage via a JTAG or I2C port. If the device allows provisioning of other certificate chain slots then it may use SPDM vendor defined messages, or some non-SPDM protocol.

**Reply by Mumunn** [reply]
OK， thank you. By the way, if SPDM Spec DSP0274_1.2.0 or above is used, whether GET_CSR and SET_CERTIFICATE CMDs must be used by device vendor as the only way to provision certificates?

**Reply by steven-bellock** [reply]
No, the device vendor is free to use non-SPDM means to provision certificate chains.

**Reply by Mumunn** [reply]
OK，thank you!
